python/lsst/analysis/ap/apdb.py

```diff
@@ -101,15 +103,19 @@ def set_excluded_diaSource_flags(self, flag_list):
 
         self.diaSource_flags_exclude = flag_list
 
-    def _make_flag_exclusion_clause(self, flag_list, column_name='flags'):
-        """Create a SQL where clause that excludes sources with chosen flags.
+    def _make_flag_exclusion_query(self, query, table, flag_list, column_name='flags'):
+        """Return an SQL where query that excludes sources with chosen flags.
 
         Parameters
         ----------
         flag_list : `list` [`str`]
             Flag names to exclude.
+        query : `sqlalchemy.orm.Query`
```

Comment:
We don't really use ORM stuff, just sqlalchemy core. The query type here should be `sqlalchemy.sql.Select`